pkg/asset/machines/powervs/machines.go

```diff
@@ -120,7 +162,32 @@ func provider(clusterID string, platform *powervs.Platform, mpool *powervs.Machi
 	return config, nil
 }
 
-// ConfigMasters sets the network and boot image IDs
-func ConfigMasters(machines []machineapi.Machine, clusterID string) {
+// ConfigMasters sets the PublicIP flag and assigns a set of load balancers to the given machines.
+func ConfigMasters(machines []machineapi.Machine, controlPlane *machinev1.ControlPlaneMachineSet, infraID string, publish types.PublishingStrategy) error {
+	lbrefs := []machinev1.LoadBalancerReference{{
+		Name: fmt.Sprintf("%s-loadbalancer-int", infraID),
+		Type: machinev1.ApplicationLoadBalancerType,
+	}}
+
+	if publish == types.ExternalPublishingStrategy {
+		lbrefs = append(lbrefs, machinev1.LoadBalancerReference{
+			Name: fmt.Sprintf("%s-loadbalancer", infraID),
+			Type: machinev1.ApplicationLoadBalancerType,
+		})
+	}
 
+	for _, machine := range machines {
+		providerSpec, ok := machine.Spec.ProviderSpec.Value.Object.(*machinev1.PowerVSMachineProviderConfig)
+		if !ok {
+			return errors.New("Unable to set load balancers to control plane machine set")
```

Comment:
nit: errors should start with a lowercase letter, but no need to change this unless you're making other changes